src/components/SideNav/SideNav.jsx

```diff
@@ -79,62 +79,31 @@ const defaultProps = {
 /**
  * Side Navigation. part of UI shell
  */
+
+const buttonBindings = [];
+
 const SideNav = ({ links, defaultExpanded, isSideNavExpanded, i18n, ...props }) => {
   const nav = links
     .map(link => {
       const enabled = link.isEnabled ? link.isEnabled : false;
       if (!enabled) {
         return null;
       }
-      if (link.hasOwnProperty('childContent')) {
-        let parentActive = false;
-        const children = link.childContent.map(childlink => {
-          if (childlink.isActive) {
-            parentActive = true;
-          }
-          return (
-            <SideNavMenuItem
-              key={`menu-link-${link.childContent.indexOf(childlink)}-child`}
-              isActive={childlink.isActive}
-              {...childlink.metaData}
-            >
-              {childlink.content}
-            </SideNavMenuItem>
-          );
-        });
-        return (
-          <SideNavMenu
-            isActive={parentActive}
-            renderIcon={link.icon}
-            aria-label="dropdown"
-            key={`menu-link-${links.indexOf(link)}-dropdown`}
-            title={link.linkContent}
-            large
-          >
-            {children}
-          </SideNavMenu>
-        );
-      }
-      return (
-        <SideNavLink
-          key={`menu-link-${link.metaData.label.replace(/\s/g, '')}-global`}
-          aria-label={link.metaData.label}
-          onClick={link.metaData.onClick}
-          href={link.metaData.href}
-          renderIcon={link.icon}
-          isActive={link.isActive}
-          {...link.metaData}
-          large
-        >
-          {link.linkContent}
-        </SideNavLink>
-      );
+      return getChildren(links, link);
     })
     .filter(i => i);
 
   const translateById = id =>
     id !== 'carbon.sidenav.state.closed' ? i18n.closeText : i18n.openText;
 
+  useEffect(() => {
+    buttonBindings.forEach(binding => {
+      binding.buttonref.current.addEventListener('click', e => {
+        binding.callback();
+      });
+    });
+  });
```

Comment:
Taylor, Thanks for the review! I am new to react :) I'll rework this a bit based on your suggestions.

Reply:
@tay1orjones I think I got it. 

I did some testing and useEffect is now only being executed once when the component loads instead of on each render. The cleanup method is also removing the listeners as expected.

I'll add a story for this and look at the @testing-library/react stuff next.

Thanks

Reply:
Also regarding your comment _I've not seen this pattern before of using a ref to bind a callback to be invoked later via an eventListener. _ Is there a cleaner way to do this in react? I'm a total noob so if I'm doing something dumb please call me out :)